typescript/packages/wallets/sui/src/SuiKeyPairWalletClient.ts

```diff
@@ -0,0 +1,112 @@
+        const signedData = await this.keypair.signPersonalMessage(messageBytes);
+        // Convert the signature to base64 string
+        return {
+            signature: signedData.toString(),
+        };
+    }
+
+    async sendTransaction(transaction: SuiTransaction): Promise<SuiTransactionResponse> {
+        const response = await this.client.signAndExecuteTransactionBlock({
+            transactionBlock: transaction.transaction,
+            signer: this.keypair,
+            options: {
+                showEffects: true,
+                showEvents: true,
+            },
+        });
+
+        return {
+            hash: response.digest,
+            digest: response.digest,
+            effects: response.effects ? {
+                status: { status: response.effects.status.status },
+                gasUsed: {
+                    computationCost: response.effects.gasUsed.computationCost,
+                    storageCost: response.effects.gasUsed.storageCost,
+                    storageRebate: response.effects.gasUsed.storageRebate,
+                },
+                transactionDigest: response.effects.transactionDigest,
+                created: response.effects.created?.map((item: OwnedObjectRef) => ({
+                    owner: typeof item.owner === 'string' ? item.owner : 'Object',
+                    reference: {
+                        objectId: item.reference.objectId,
+                        version: item.reference.version,
+                        digest: item.reference.digest,
+                    },
+                })),
+                mutated: response.effects.mutated?.map((item: OwnedObjectRef) => ({
+                    owner: typeof item.owner === 'string' ? item.owner : 'Object',
+                    reference: {
+                        objectId: item.reference.objectId,
+                        version: item.reference.version,
+                        digest: item.reference.digest,
+                    },
+                })),
+                deleted: response.effects.deleted?.map((item: SuiObjectRef) => ({
+                    owner: "Deleted",
+                    reference: {
+                        objectId: item.objectId,
+                        version: item.version,
+                        digest: item.digest,
+                    },
+                })),
+            } : undefined,
+            events: response.events?.map(event => ({
+                id: event.id.toString(),
+                packageId: event.packageId,
+                transactionModule: event.transactionModule,
+                sender: event.sender,
+                type: event.type,
+                parsedJson: event.parsedJson as Record<string, unknown>,
+                bcs: event.bcs,
+                timestampMs: event.timestampMs ? Number(event.timestampMs) : undefined,
+            })),
+        };
+    }
+
+    async read(query: SuiQuery): Promise<SuiQueryResponse> {
```

Comment:
This is incorrect. Re-implement the read function and interface in a way that i can read any Sui contract